### DISCUSSION TITLE
setting up the single-user database

### DISCUSSION THREAD

**Discussion by JBSmo** [rffrasca/PDFKeeper#14] [General]
I am evaluating your software to see if it will work for a repository of PDF documents that were created by a CAD program. Although I'm evaluating it on my personal computer, the intended end use is on a computer with no internet access whatsoever. To that end, I've tried downloading and then running the SingleUserDatabaseSetup windows command script, which fails with "the system cannot find the file specified".

I figure this is the SqliteDatabaseSetup.sql file. Is that correct? Is it possible to get a local copy of this file?

Perhaps a larger (and more pertinent) question would be: Can this software be used on such an isolated system?

Thanks for your attention!

**Comment by rffrasca** [comment]
The scripts needed for the database setup are included with PDFKeeper and are designed to be called from within the help file.
On first launch of PDFKeeper, the help file will open:
1. From "Getting Started", select "Setup Single-User Database."
2. Select "Setup Single-User Database Now!"
3. Select Run when prompted
4. Select "y" to create the database

To answer your question, PDFKeeper should be able to operate offline; however, on startup and every 60 minutes, there is an update check. If that poses a problem, you can create an issue for it, and I will create a setting to disable.

**Comment by JBSmo** [comment]
Thanks for your prompt response, Robert. 

Yes, I followed step 1 and 2, above, but thought the command script was being executed from an online repository, so I saved it to my downloads folder and ran it from there. That's when I ran into the cannot find file problem.

To continue my evaluation, I DID follow step 3 and 4 on a subsequent try and was able to open the application and add some PDF files.
This raises an additional question, however. Is there a mechanism for editing existing records, in case I forgot to fill in, or made a mistake in, one of the fields on the add form?

As for the update check - if it fails to connect what is the behavior? I can live with the occasional (every hour or so) error message telling me the application can't "phone home". If it prevents the application from running, that would be an obvious problem.

Thanks again!

**Reply by rffrasca** [reply]
I tested running PDFKeeper offline. I installed the previous version (8.1.1), dropped my connection, and then launched PDFKeeper. The auto update app didn't popup on startup.

As for editing existing records, Category and Tax Year can be edited but, Title, Author, and Subject can only be edited with SQL. If you don't mind, please open an enhancement and I will work it into a future release.

**Comment by JBSmo** [comment]
Thanks again, Robert. I added a feature request as you suggested.

**Reply by rffrasca** [reply]
You're welcome!
I will also add a note to the help file that will instruct the user to "Select Run when prompted".

**Comment by rffrasca** [comment]
Issue #18 was created for the updating of the help file to instruct the user to select Run when prompted.
